mindshit.py

```diff
@@ -187,105 +192,15 @@ def lex(self) -> Tuple[List[Token], Error]:
-                tokens.append(Token(Tk.OP, '==', self.pos))
-                self.next(2)
-                
-            elif self.chars(2) == '!=':
-                tokens.append(Token(Tk.OP, '!=', self.pos))
-                self.next(2)
-                
-            elif self.chars(2) == '>=':
-                tokens.append(Token(Tk.OP, '>=', self.pos))
-                self.next(2)
-                
-            elif self.chars(2) == '<=':
-                tokens.append(Token(Tk.OP, '<=', self.pos))
-                self.next(2)
-                
-            elif self.char == '&':
-                tokens.append(Token(Tk.OP, '&', self.pos))
-                self.next()
-            
-            elif self.char == '=':
-                tokens.append(Token(Tk.OP, '=', self.pos))
-                self.next()
-            
-            elif self.char == ':':
-                tokens.append(Token(Tk.OP, ':', self.pos))
-                self.next()
-            
-            elif self.char == '(':
-                tokens.append(Token(Tk.OP, '(', self.pos))
-                self.next()
-                
-            elif self.char == ')':
-                tokens.append(Token(Tk.OP, ')', self.pos))
-                self.next()
-            
-            elif self.char == '[':
-                tokens.append(Token(Tk.OP, '[', self.pos))
-                self.next()
-                
-            elif self.char == ']':
-                tokens.append(Token(Tk.OP, ']', self.pos))
-                self.next()
-                
-            elif self.char == '+':
-                tokens.append(Token(Tk.OP, '+', self.pos))
-                self.next()
-                
-            elif self.char == '-':
-                tokens.append(Token(Tk.OP, '-', self.pos))
-                self.next()
-                
-            elif self.char == '*':
-                tokens.append(Token(Tk.OP, '*', self.pos))
-                self.next()
-                
-            elif self.char == '/':
-                tokens.append(Token(Tk.OP, '/', self.pos))
-                self.next()
-                
-            elif self.char == '%':
-                tokens.append(Token(Tk.OP, '%', self.pos))
-                self.next()
-                
-            elif self.char == '>':
-                tokens.append(Token(Tk.OP, '>', self.pos))
-                self.next()
-                
-            elif self.char == '<':
-                tokens.append(Token(Tk.OP, '<', self.pos))
-                self.next()
             
             else:
-                return [], IllegalCharError(f"'{self.char}'", self.pos)
+                ok = False
+                for valid_token in valid_tokens:
+                    if self.chars(len(valid_token)) == valid_token:
+                        ok = True
+                        parseToken(valid_token, len(valid_token))
+                if not ok:
+                    return [], IllegalCharError(f"'{self.char}'", self.pos)
```

Comment:
Hopefully this works
Actually, it should break the for loop if `self.chars(len(valid_token)) == valid_token`, since its redundant to check tokens when I've already matched one of 'em